### PR TITLE
add message log to asp.net

### PR DESCRIPTION
**Pull request by wanlwanl**
### Summary of the changes (Less than 80 chars)
 - message arrive server from diagnostic client
 - message send to service from server

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs [hunk 1]
```diff
@@ -17,14 +18,16 @@ internal class ServiceMessageBus : MessageBus
...
 
-        public ServiceMessageBus(IDependencyResolver resolver) : base(resolver)
+        public ServiceMessageBus(IDependencyResolver resolver, ILogger<ServiceMessageBus> logger) : base(resolver)
         {
             // TODO: find a more decent way instead of DI, it can be easily overriden
             _serviceConnectionManager = resolver.Resolve<IServiceConnectionManager>() ?? throw new ArgumentNullException(nameof(IServiceConnectionManager));
             _clientConnectionManager = resolver.Resolve<IClientConnectionManager>() ?? throw new ArgumentNullException(nameof(IClientConnectionManager));
             _parser = resolver.Resolve<IMessageParser>() ?? throw new ArgumentNullException(nameof(IMessageParser));
             _ackHandler = resolver.Resolve<IAckHandler>() ?? throw new ArgumentNullException(nameof(IAckHandler));
+            _logger = logger ?? throw new ArgumentNullException(nameof(ILogger<ServiceMessageBus>));
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:30]: `nameof(logger)`

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/SignalRMessageParser.cs [hunk 2]
```diff
@@ -60,7 +63,11 @@ public IEnumerable<AppMessage> GetMessages(Message message)
                             var groupName = command.Value;
                             var connectionId = GetName(message.Key, PrefixHelper.ConnectionIdPrefix);
                             var joinGroupWithAckMessage = new JoinGroupWithAckMessage(connectionId, groupName).WithTracingId();
-                            
+                            if (joinGroupWithAckMessage.TracingId != null)
+                            { 
+                                AzureSignalRLog.StartToAddConnectionToGroup(_logger, joinGroupWithAckMessage);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.AspNet/HubHost/SignalRMessageParser.cs:68]: start doing sth....

File: src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@ internal static class AzureSignalRLog
         public const string StartToBroadcastMessageTemplate = "Start to broadcast message {0}.";
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs:14]: the class name, maybe simply `Logger`

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs [hunk 4]
```diff
@@ -43,14 +46,31 @@ public override Task Publish(Message message)
...
-        private Task ProcessMessage(AppMessage message)
+        private async Task ProcessMessage(AppMessage message)
         {
             if (message is HubMessage hubMessage)
             {
-                return WriteMessage(_serviceConnectionManager.WithHub(hubMessage.HubName), message);
+                try
+                {
+                    await WriteMessage(_serviceConnectionManager.WithHub(hubMessage.HubName), message);
+                    if (hubMessage.Message is IMessageWithTracingId msg && msg.TracingId != null)
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: discussion: is moving the check into Log better?

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/SignalRMessageParser.cs (hunk 2 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/SignalRMessageParser.cs:68]: looks like both `start to do` and `start doing` are valid https://ell.stackexchange.com/questions/105862/started-to-do-and-started-doing

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs (hunk 1 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:30]: I just keep the conventions... look at the previous lines from L26-L29

File: src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs (hunk 3 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs:14]: Sounds good

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs (hunk 4 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: I don't think so. Actually I try to hide the check inside Log in RT, but later I need to modify some check in one of the callers, and I am not sure in what condition I log it if I don't view the definition of the log, not convenient if you consider it.

Comment (KKhurin) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: Can the message be `IMessageWithTracingId` but its `TracingId` be null?

File: src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs (hunk 3 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Common/Logging/AzureSignalRLog.cs:14]: but a lot of classes already has a menber called `Logger`, they will be conflict

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs [hunk 5]
```diff
@@ -43,14 +46,31 @@ public override Task Publish(Message message)
...
+                    {
+                        AzureSignalRLog.FailedToSendMessage(_logger, msg, ex);
+                    }
+                    throw;
+                }
             }
 
-            return WriteMessage(_serviceConnectionManager, message);
+            await WriteMessage(_serviceConnectionManager, message);
+            return;
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:73]: remove redundant `return`

File: src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs (hunk 4 above)
Comment (vicancy) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: yeah, maybe move the `TracingId` null check into Log? BTW, can message not `IMessageWithTracingId`? it feels annoying to do some if check before Logging everytime

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: that's valid for a message, we only need to log in 2 cases:
* message sent from diagnostic client
* the service enable messaing log
Before messages reaches here, the tracing ID is already generated (in ServiceLifetimeManagerBase). 
So we only need to log the message with non-null tracing ID

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.AspNet/HubHost/ServiceMessageBus.cs:56]: if the message isn't `IMessageWithTracingId`, I won't log it. For those messages I'm insterested, I already implement the `IMessageWithTracingId`

**Review by vicancy — APPROVED** [review]

**Comment by wanlwanl** [comment]
Discussed offline with @vicancy 
todo in next PR:
* move check into `AzureSignalRLog`
* rename `AzureSignalRLog`

**Pull request merged**